libs/remote_storage/src/azure_blob.rs

```diff
@@ -174,6 +174,15 @@ impl AzureBlobStorage {
                     .map_err(|e| DownloadError::Other(e.into()))?;
                 bufs.push(data);
             }
+
+            if bufs.is_empty() {
+                return Err(DownloadError::Other(anyhow::anyhow!(
+                    "Azure GET response contained no buffers"
+                )));
+            }
+            let etag = etag.unwrap();
+            let last_modified = last_modified.unwrap();
```

Comment:
This `unwrap` isn't for missing data on the response: it's safe because if bufs was empty, we would already have returned an error, and if bufs is non-empty then these are definitely set.  I'll add a comment to that effect.